test/net_http_test.rb

```diff
@@ -0,0 +1,218 @@
+require 'test_helper'
+require 'semian/net_http'
+require 'thin'
+
+class TestNetHTTP < MiniTest::Unit::TestCase
+  class RackServer
+    def self.call(env)
+      response_code = env['REQUEST_URI'].delete("/")
+      response_code = '200' if response_code == ""
+      [response_code, {'Content-Type' => 'text/html'}, ['Success']]
+    end
+  end
+
+  def setup
+    # create a simple rack server so we can use toxiproxy
+    start_server_singleton
+    Semian["http_localhost_41234"].reset if Semian["http_localhost_41234"]
+    Semian["http_localhost_41235"].reset if Semian["http_localhost_41235"]
+    Semian.destroy('http_localhost_41234')
+    Semian.destroy('http_localhost_41235')
+    @proxy = Toxiproxy[:semian_test_net_http]
+  end
+
+  def test_server_up_and_toxiproxy_working
+    uri = URI('http://localhost:41234/200')
+    assert_equal "Success", Net::HTTP.get(uri)
+
+    uri = URI('http://localhost:41235/200')
+    @proxy.downstream(:latency, latency: 300).apply do
+      time = Time.now
+      Net::HTTP.get(uri)
+
+      time_diff = Time.now - time
+      assert time_diff >= 0.3
+    end
+  end
+
+  def test_semian_identifier
+    Net::HTTP.start("localhost", 41_235) do |http|
+      assert_equal "http_localhost_41235", http.semian_identifier
+    end
+  end
+
+  def test_trigger_open
+    open_circuit!
+    uri = URI('http://localhost:41235/200')
+    assert_raises Net::CircuitOpenError do
+      Net::HTTP.get(uri)
+    end
+  end
+
+  def test_trigger_close_after_open
+    open_circuit!
+    close_circuit!
+  end
+
+  def test_blockheads_tickets_are_working
+    threads = []
+    ticket_count = Net::HTTP.new("localhost", 41_235).raw_semian_options[:tickets]
+
+    @proxy.downstream(:latency, latency: 500).apply do
+      uri = URI('http://localhost:41235/200')
+      (ticket_count).times do
+        threads << Thread.new do
+          Net::HTTP.get(uri)
+        end
+      end
+      sleep 0.3
```

Comment:
I have to keep at least `ticket_count` requests blocked while running a `ticket_count+1`th request. The 500 ms latency keeps it blocked while after 300 ms the `ticket_count+1`th request runs.